test/e2e/cert_manager_deployment_test.go

```diff
@@ -52,3 +55,76 @@ func TestSelfSignedCerts(t *testing.T) {
 	})
 	require.NoError(t, err)
 }
+
+func TestACMECertsIngress(t *testing.T) {
+	loader := library.NewDynamicResourceLoader(context.TODO(), t)
+	config, err := library.GetConfigForTest(t)
+	require.NoError(t, err)
+
+	loader.CreateFromFile(testassets.ReadFile, filepath.Join("testdata", "acme", "clusterissuer.yaml"))
+	defer loader.DeleteFromFile(testassets.ReadFile, filepath.Join("testdata", "acme", "clusterissuer.yaml"))
+	loader.CreateFromFile(testassets.ReadFile, filepath.Join("testdata", "acme", "hello.yaml"))
+	defer loader.DeleteFromFile(testassets.ReadFile, filepath.Join("testdata", "acme", "hello.yaml"))
+
+	routeV1Client, err := routev1.NewForConfig(config)
+	require.NoError(t, err)
+	route, err := routeV1Client.Routes("openshift-console").Get(context.TODO(), "console", metav1.GetOptions{})
+	require.NoError(t, err)
+
+	ingress_host := "hello-openshift." + strings.Join(strings.Split(route.Spec.Host, ".")[1:], ".")
+	path_type := networkingv1.PathTypePrefix
+	ingress := &networkingv1.Ingress{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "networking.k8s.io/v1",
+			Kind:       "Ingress",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "ingress-le-prod",
+			Namespace: "hello",
+			Annotations: map[string]string{
+				"cert-manager.io/cluster-issuer": "letsencrypt-prod",
+			},
+		},
+		Spec: networkingv1.IngressSpec{
+			Rules: []networkingv1.IngressRule{
+				{
+					Host: ingress_host,
+					IngressRuleValue: networkingv1.IngressRuleValue{
+						HTTP: &networkingv1.HTTPIngressRuleValue{
+							Paths: []networkingv1.HTTPIngressPath{
+								{
+									Path:     "/",
+									PathType: &path_type,
+									Backend:  networkingv1.IngressBackend{Service: &networkingv1.IngressServiceBackend{Name: "hello-openshift", Port: networkingv1.ServiceBackendPort{Number: 8080}}},
+								},
+							},
+						},
+					},
+				},
+			},
+			TLS: []networkingv1.IngressTLS{{
+				Hosts:      []string{ingress_host},
+				SecretName: "ingress-prod-secret",
+			}},
+		},
+	}
+	ingress, err = loader.KubeClient.NetworkingV1().Ingresses(ingress.ObjectMeta.Namespace).Create(context.TODO(), ingress, metav1.CreateOptions{})
+	require.NoError(t, err)
```

Comment:
Since we have a working Ingress at this stage, could we just call it using some HTTP client and verify if the certificate issuer looks fine? This way, the test will be more like a real e2e thing.

Reply:
That is what i'm working on right now :+1: